Microsoft.WindowsAzure.Storage/tests/blob_streams_test.cpp

```diff
@@ -178,6 +179,60 @@ void seek_and_write_putn(concurrency::streams::ostream stream, const std::vector
 
 SUITE(Blob)
 {
+	TEST_FIXTURE(block_blob_test_base, blob_write_stream_upload_and_download)
+	{
+        azure::storage::core::hash_provider provider = azure::storage::core::hash_provider::create_md5_hash_provider();
+
+        std::vector<uint8_t> buffer;
+        size_t buffersize = 3 * 1024 * 1024;
+        buffer.resize(buffersize);
+        
+        auto wstream = m_blob.open_write();
+
+        std::generate_n(buffer.begin(), buffer.size(), []() -> uint8_t
+        {
+            return (uint8_t)(std::rand() % (int)UINT8_MAX);
+        });
+        
+        provider.write(buffer.data(), buffersize);
+        
+        concurrency::streams::container_buffer<std::vector<uint8_t>> input_buffer(buffer);
+        
+        wstream.write(input_buffer, buffersize);
+        
+        std::generate_n(buffer.begin(), buffer.size(), []() -> uint8_t
+        {
+            return (uint8_t)(std::rand() % (int)UINT8_MAX);
+        });
+        provider.write(buffer.data(), buffersize);
+        input_buffer = concurrency::streams::container_buffer<std::vector<uint8_t>>(buffer);
+        wstream.write(input_buffer, buffersize);
+
+        std::generate_n(buffer.begin(), buffer.size(), []() -> uint8_t
+        {
+            return (uint8_t)(std::rand() % (int)UINT8_MAX);
+        });
+        provider.write(buffer.data(), buffersize);
+        input_buffer = concurrency::streams::container_buffer<std::vector<uint8_t>>(buffer);
+        wstream.write(input_buffer, buffersize);
+
```

Comment:
Why did you repeat the same block of code three times? Lines 192-199 generate `buffersize` octets of random data; lines 203-208 and 211-216 are literally identical (except line 192 declares-and-sets while lines 203 and 211 just set). Those lines should be extracted to a helper function which take `buffer` and `buffersize` as parameters and returns a `concurrency::streams::container_buffer<std::vector<uint8_t>>`.

Reply:
And now you have three much cleaner blocks of code... that are now clearly and obviously identical. Is there a reason they're not rolled up into a loop?

Oh, and the indentation appears to be pretty badly broken in this function; some lines are flush-left without indentation at all.